src/resolve/resolved-dns-stub.c

```diff
@@ -801,16 +802,18 @@ static void dns_stub_process_query(Manager *m, DnsStubListenerExtra *l, DnsStrea
                 return;
         }
 
-        if (dns_type_is_obsolete(p->question->keys[0]->type)) {
-                log_debug("Got message with obsolete key type, refusing.");
-                dns_stub_send_failure(m, l, s, p, DNS_RCODE_REFUSED, false);
-                return;
-        }
+        DNS_QUESTION_FOREACH(key, p->question) {
```

Comment:
the dns stub only answers classic DNS, and on classic DNS there must be exactly 1 key in the question section, hence no need for the loop. (and DNS questions with multiple questions should already have been refused while parsing, further up) this should be changed to you dns_question_first_key() though...